types/steam-tradeoffer-manager/lib/classes/TradeOffer.d.ts

```diff
@@ -381,7 +372,7 @@ declare class TradeOffer {
      * @param callback - Required. A callback to be invoked when complete.
      * - `err` - An `Error` object on failure, `null` on success
      */
-    update(callback: (err: Error | null) => void): void;
+    update(callback: (err: TradeOfferManager.EResultError | null) => void): void;
```

Comment:
Thank you for discovering this bug, but this actually needs to be changed to the `Error & { eresult?: EResult }` _without_ a cause because `update` never puts a cause but _may_ put an eresult.

https://github.com/DoctorMcKay/node-steam-tradeoffer-manager/blob/master/lib/classes/TradeOffer.js#L577
https://github.com/DoctorMcKay/node-steam-tradeoffer-manager/blob/master/lib/webapi.js#L56

Reply:
I did not notice that. I think it will be best to make the EResultError type just contain the eresult and then create a second type called EResultErrorWithCause for errors that might contain a cause.

Reply:
Actually I see why you did it the way you did it with the cause. I will not be changing the causes but instead just fix the problem I had to begin with where the send function and update function did not include eresult in the error.